apps/s_server.c

```diff
@@ -1814,11 +1814,21 @@ int s_server_main(int argc, char *argv[])
     if (bio_s_out == NULL) {
         if (s_quiet && !s_debug) {
             bio_s_out = BIO_new(BIO_s_null());
-            if (s_msg && bio_s_msg == NULL)
+            if (s_msg && bio_s_msg == NULL) {
                 bio_s_msg = dup_bio_out(FORMAT_TEXT);
+                if (bio_s_msg == NULL) {
+                    BIO_printf(bio_err, "Out of memory\n");
+                    goto end;
+                }
+            }
         } else {
             bio_s_out = dup_bio_out(FORMAT_TEXT);
         }
+
+        if (bio_s_out == NULL) {
```

Comment:
There is already a test for this below

Reply:
Sorry, I did not notice this when resolving conflicts with https://github.com/openssl/openssl/pull/17710.